config/backend-config/multitenant_workspaces.go

```diff
@@ -11,19 +11,28 @@ import (
 	"github.com/rudderlabs/rudder-server/config"
 )
 
-type MultiTenantWorkspaceConfig struct {
+type MultiTenantWorkspacesConfig struct {
 	CommonBackendConfig
+	Token                     string
 	writeKeyToWorkspaceIDMap  map[string]string
 	sourceToWorkspaceIDMap    map[string]string
 	workspaceIDToLibrariesMap map[string]LibrariesT
 	workspaceWriteKeysMapLock sync.RWMutex
 }
 
-func (workspaceConfig *MultiTenantWorkspaceConfig) SetUp() {
+func (workspaceConfig *MultiTenantWorkspacesConfig) SetUp() {
 	workspaceConfig.writeKeyToWorkspaceIDMap = make(map[string]string)
+
+	if workspaceConfig.Token == "" {
+		workspaceConfig.Token = config.GetEnv("HOSTED_MULTITENANT_SERVICE_SECRET", "password")
```

Comment:
Why do we default to `password`? Also, is it valid if this setting is not provided?

Reply:
Good point, I just moved this without thinking about it. I believe if the setting is missing we should error.

Reply:
done